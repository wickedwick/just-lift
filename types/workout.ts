export type Exercise = {
  id: string,
  name: string
  sets: number
  reps: number
  weight: number
  weightUnit: WeightUnit
  progressiveOverload: boolean
  overloadIncrement: number
}

export type Workout = {
  id: string
  exercises: Exercise[]
}

export type WorkoutPlan = {
  workouts: Workout[]
  daysPerWeek: number
  workoutIndex: number
  workoutInProgress: boolean
}

export type Log = {
  date: Date
  workoutId: string
  data: string
  exerciseName: string
  weight: number
  weightUnit: WeightUnit
}

export enum WeightUnit {
  Lbs,
  Kg
}
 