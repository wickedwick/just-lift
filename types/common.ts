import { Exercise, Log, WorkoutPlan } from './workout';
import { StyleProp, TextStyle, ViewStyle } from 'react-native';

export type RootStackParamList = {
  Root: undefined,
  NotFound: undefined
}

export type BottomTabParamList = {
  AutoTagger: undefined
  Celebrities: undefined
}

export type TabOneParamList = {
  BlankMenuScreen: undefined
  CreateWorkoutScreen: undefined
  SelectWorkoutScreen: undefined
  WorkoutScreen: undefined
  SettingsScreen: undefined
  ProfileScreen: undefined
  GuidedSelectionScreen: undefined
  WorkoutListScreen: undefined
}

export type TabTwoParamList = {
  TabOneScreen: undefined
  TabTwoScreen: undefined
}

export type WorkoutFormProps = {
  exercise?: Exercise
  onSubmit: (exercise: Exercise) => void
  onCancel: () => void
}

export type ExerciseCardProps = {
  exercise: Exercise
  onRemovePress: () => void
  onEditPress: () => void
}

export type ExerciseCounterProps = {
  exercise: Exercise
  setLogData: (log: Log) => void
  setWeight: (weight: number, exerciseName: string) => void
}

export type LogCardProps = {
  log: Log
}

export type ActionButtonProps = {
  onPress: () => void
  contained?: boolean
  icon?: string
  labelStyle?: StyleProp<TextStyle>
  style?: StyleProp<ViewStyle>
  text: string
}

export type RadioButtonProps = {
  label: string
  value: string
}

export type CheckboxProps = {
  label: string
  status: 'checked' | 'unchecked'
  onPress: () => void
}

export type WorkoutPlanContextProps = {
  workoutPlan: WorkoutPlan | null
  setWorkoutPlan: (workoutPlan: WorkoutPlan | null) => void
}
