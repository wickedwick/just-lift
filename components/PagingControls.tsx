import ActionButton from './ActionButton';
import React from 'react';
import { PagingControlsProps } from '../types/common';
import { StyleSheet, Text, View } from 'react-native';
import { useTheme } from 'react-native-paper';

const PagingControls = (props: PagingControlsProps): JSX.Element => {
  const { workoutIndex, workouts, setWorkoutIndex } = props
  const { colors } = useTheme()
  const styles = createStyles(colors)
  
  return (
    <View style={styles.flexContainer}>
      {workoutIndex > 0 && (
        <ActionButton
          contained
          onPress={() => { setWorkoutIndex(workoutIndex - 1) }}
          text="Previous"
        />
      )}

      {workouts && <Text style={styles.heading}>Workout {workouts[workoutIndex]?.id}</Text>}
      
      {workoutIndex < workouts.length - 1 && (
        <ActionButton
          contained
          onPress={() => { setWorkoutIndex(workoutIndex + 1) }}
          text="Next"
        />
      )}
    </View>
  )
}

const createStyles = (colors: ReactNativePaper.ThemeColors) => StyleSheet.create({
  flexContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  heading: {
    fontSize: 20,
    fontWeight: 'bold',
    marginTop: 5,
    color: colors.text,
  }
})

export default PagingControls
