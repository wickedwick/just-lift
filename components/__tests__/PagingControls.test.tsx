import Adapter from 'enzyme-adapter-react-16'
import PagingControls from '../PagingControls'
import React from 'react'
import { configure, shallow } from 'enzyme'
import { PagingControlsProps } from '../../types/common'

configure({ adapter: new Adapter() })

const mockSetWorkoutIndex = jest.fn()
const props: PagingControlsProps = {
  workoutIndex: 0,
  workouts: [{ id: 'A', exercises: [] }, { id: 'B', exercises: [] }],
  setWorkoutIndex: mockSetWorkoutIndex,
}

describe('<PagingControls />', () => {
  it('Renders', () => {
    const wrapper = shallow(<PagingControls {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Calls setWorkoutIndex on press', () => {
    const wrapper = shallow(<PagingControls {...props} />)
    wrapper.find('ActionButton').at(0).simulate('press')
    expect(mockSetWorkoutIndex).toHaveBeenCalledWith(1)
  })
})
