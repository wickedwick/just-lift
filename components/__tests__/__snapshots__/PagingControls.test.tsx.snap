// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationControls /> Renders 1`] = `ShallowWrapper {}`;
